src/chain/chain-training.h

```diff
@@ -61,8 +61,18 @@ struct ChainTrainingOptions {
   // should have a softmax as its final nonlinearity.
   BaseFloat xent_regularize;
 
+  // The boosting coefficient (i.e. b in bMMI)
+  BaseFloat boost;
+  
+  // This controls the boosting approach. If true, hard boosting
+  // will be used: the boosting mask consists of 1's and 0's and is 
+  // independent of numerator compuatation (i.e. is a function of numerator
```

Comment:
I'm not familiar with boosted MMI, but do you mean "independent of denominator computation"?